apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/cache.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2016 Cisco Systems Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from keystoneclient import auth as ksc_auth
+from keystoneclient import session as ksc_session
+from keystoneclient.v3 import client as ksc_client
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from neutron._i18n import _LW
+
+LOG = logging.getLogger(__name__)
+
+# REVISIT: We use keystone to get the name of the keystone
+# project owning each neutron resource, which by default, requires
+# admin. If we keep this, we should probably move it to a separate
+# config module. But we should also investigate whether admin is even
+# needed, or if neutron's credentials could somehow be used.
+AUTH_GROUP = 'apic_aim_auth'
```

Comment:
I think there should not be references to 'aim' in this code at all. Also the commit message mentions the name of this section to "apic_ml2_auth" but the code uses "apic_aim_auth"

Reply:
Mandeep and I discussed this. The naming is unfortunate, as we've already had a release with this in our release notes (i.e. we thought we were already using this for v3 keystone config across all variants -- it ends up the legacy apic-ml2-driver wasn't using it). The decision was to keep the same name in order to keep this consistent.

Reply:
Will fix apic_ml2_auth -- my original patch changed it to that name, before Mandeep and I had discussed it.